data-raw/vs_metabolic.R

```diff
@@ -0,0 +1,295 @@
+          VISIT == "WEEK 20" ~ 96.95,
+          VISIT == "WEEK 24" ~ 94.86,
+          VISIT == "WEEK 26" ~ 94.11
+        ),
+        USUBJID == "01-701-1034" & # Insulin
+          VSTESTCD == "HEIGHT" ~ case_when(VISIT == "SCREENING 1" ~ 163),
+        USUBJID == "01-701-1034" & # Insulin
+          VSTESTCD == "WSTCIR" ~ case_when(
+          VISIT == "SCREENING 1" ~ 115,
+          VISIT == "BASELINE" ~ 115,
+          VISIT == "WEEK 2" ~ 114,
+          VISIT == "WEEK 4" ~ 113,
+          VISIT == "WEEK 6" ~ 112,
+          VISIT == "WEEK 8" ~ 113,
+          VISIT == "WEEK 12" ~ 110,
+          VISIT == "WEEK 16" ~ 109.2,
+          VISIT == "WEEK 20" ~ 109,
+          VISIT == "WEEK 24" ~ 108,
+          VISIT == "WEEK 26" ~ 107
+        ),
+        USUBJID == "01-701-1034" & # Insulin
+          VSTESTCD == "HIPCIR" ~ case_when(
+          VISIT == "SCREENING 1" ~ 120,
+          VISIT == "BASELINE" ~ 119.7,
+          VISIT == "WEEK 2" ~ 119.1,
+          VISIT == "WEEK 4" ~ 118.3,
+          VISIT == "WEEK 6" ~ 117.7,
+          VISIT == "WEEK 8" ~ 117,
+          VISIT == "WEEK 12" ~ 116.4,
+          VISIT == "WEEK 16" ~ 115,
+          VISIT == "WEEK 20" ~ 114,
+          VISIT == "WEEK 24" ~ 113,
+          VISIT == "WEEK 26" ~ 113
+        ),
+        TRUE ~ VSSTRESN
+      )
+  )
+
+# Derive BMI values ----
+
+vs_mb_bmi <- vs_mb |>
+  # use body weight as a base for the table
+  filter(VSTESTCD == "WEIGHT") |>
+  mutate(VSSTRESN = as.numeric(VSSTRESN)) |>
+  # rename value to body weight
+  rename(BODYWEIGHT = VSSTRESN) |>
+  # drop irrelevant columns
+  select(-VSORRES, -VSORRESU) |>
+  # add height
+  left_join(
+    vs_mb |>
+      # pick height
+      filter(VSTESTCD == "HEIGHT") |>
+      # Convert height from cm to m
+      mutate(VSSTRESN = as.numeric(VSSTRESN / 100)) |>
+      # rename
+      rename(HEIGHT = VSSTRESN) |>
+      # drop irrelevant
+      select(USUBJID, HEIGHT),
+    by = "USUBJID"
+  ) |>
+  # calculate bmi
+  mutate(
+    VSSTRESN = as.numeric(BODYWEIGHT / (HEIGHT^2)),
+    VSTESTCD = "BMI",
+    VSTEST = "Body Mass Index",
+  ) |>
+  # drop irrelevant
+  select(-BODYWEIGHT, -HEIGHT)
+
+# Bind BMI dataset to vs_mb dataset
+vs_mb <- bind_rows(vs_mb, vs_mb_bmi) %>% arrange(USUBJID, VSTESTCD, VISITNUM, VSDY)
+
+# Formatting the output dataset ----
+
+vs_mb <- vs_mb %>%
+  mutate(
+    VSSEQ = row_number(),
+    VSORRESU = case_when(
+      VSTESTCD == "WEIGHT" ~ "kg",
+      VSTESTCD == "BMI" ~ "kg/m2",
+      VSTESTCD == "WSTCIR" ~ "cm",
+      VSTESTCD == "HIPCIR" ~ "cm",
+      VSTESTCD == "HEIGHT" ~ "cm",
+      TRUE ~ VSORRESU
+    ),
+    VSSTRESC = as.character(VSSTRESN),
+    VSORRES = VSSTRESC,
+    VSSTRESU = VSORRESU,
+  ) %>%
+  arrange(USUBJID, VSTESTCD, VISITNUM, VSDY)
+# get common column names
+
+common_cols <- intersect(names(vs_mb), names(vs))
+
+# assign labels to mb_vs
+
+walk(common_cols, \(x) attr(vs_mb[[x]], "label") <<- attr(vs[[x]], "label"))
+
```

Comment:
Can you reorder the `vs_metabolic` columns so that they are in the same order as for `vs`?